src/v/utils/rjson_allocator.h

```diff
@@ -0,0 +1,26 @@
+#include <rapidjson/allocators.h>
+#include <rapidjson/document.h>
+#include <rapidjson/encodings.h>
+
+namespace rjson_allocator {
+class MyAllocator : public rapidjson::CrtAllocator {
+public:
+    void* Malloc(size_t size) {
+        if (!size) {
+            return NULL;
+        }
```

Comment:
This is already checked by `CrtAllocator`, it's probably best to defer responsibility.